### PR TITLE
[SPARK-39577][SQL][DOCS] Add SQL reference for built-in functions

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Currently, Spark SQL reference missing many functions.
Users cannot find the needed functions.

### Why are the changes needed?
Add SQL reference for built-in functions

### Does this PR introduce _any_ user-facing change?
'Yes'.
Users can find needed functions in SQL reference.
Before this PR, the built-in functions show below.
image

After this PR, the built-in functions show below.
image

The part of Mathematical Functions show below.
image
image
image
image

The part of String Functions show below.
image
image
image
image
image
image
image

The part of Bitwise Functions show below.
image

The part of Conversion Functions show below.
image

The part of Conditional Functions show below.
image

The part of Predicate Functions show below.
image
image

The part of Csv Functions show below.
image

The part of Misc Functions show below.
image
image

The part of Generator Functions show below.
image

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by beliefer** [comment]
ping @HyukjinKwon cc @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
some nit:
1. `* expr2` looks weird, shall we use `expr1 * expr2`?
2. aren't math funtions also scalar functions?
3. can we include all the new content in the screenshot?

**Comment by beliefer** [comment]
> some nit:
> 
> 1. `* expr2` looks weird, shall we use `expr1 * expr2`?
> 2. aren't math funtions also scalar functions?
> 3. can we include all the new content in the screenshot?

Updated.

File: docs/sql-ref-functions-builtin.md [hunk 1]
```diff
@@ -77,3 +77,93 @@ license: |
...
+{% include_relative generated-predicate-funcs-table.html %}
+#### Examples
+{% include_relative generated-predicate-funcs-examples.html %}
+        {% break %}
+    {% endif %}
+{% endfor %}
+
+{% for static_file in site.static_files %}
+    {% if static_file.name == 'generated-generator-funcs-table.html' %}
+### Generator Functions
```
Comment (cloud-fan) [inline docs/sql-ref-functions-builtin.md:142]: This is not scalar function and should be put in a new section. Shall we do it in a new PR?

Comment (beliefer) [reply docs/sql-ref-functions-builtin.md:142]: OK

**Comment by cloud-fan** [comment]
thanks, merging to master/3.3!

**Comment by beliefer** [comment]
@cloud-fan Thank you !

**Pull request closed without merging**